tests/unit/test_ops.py

```diff
@@ -406,3 +406,129 @@ def test_dropna(tmpdir, datasets, engine="parquet"):
         new_gdf = dropna.apply_op(gdf, columns_ctx, "all")
         assert new_gdf.columns.all() == gdf.columns.all()
         assert new_gdf.isnull().all().sum() < 1, "null values exist"
+
+
+@pytest.mark.parametrize("gpu_memory_frac", [0.01, 0.1])
+@pytest.mark.parametrize("engine", ["parquet"])
+def test_seriesops(tmpdir, datasets, gpu_memory_frac, engine):
+    paths = glob.glob(str(datasets[engine]) + "/*." + engine.split("-")[0])
+
+    if engine == "parquet":
+        df1 = cudf.read_parquet(paths[0])[allcols_csv]
+        df2 = cudf.read_parquet(paths[1])[allcols_csv]
+    else:
+        df1 = cudf.read_csv(paths[0], header=False, names=allcols_csv)[allcols_csv]
+        df2 = cudf.read_csv(paths[1], header=False, names=allcols_csv)[allcols_csv]
+    df = cudf.concat([df1, df2], axis=0)
+    df["id"] = df["id"].astype("int64")
+
+    if engine == "parquet":
+        columns = allcols_csv
+    else:
+        columns = allcols_csv
+
+    data_itr = nvtabular.io.GPUDatasetIterator(
+        paths,
+        columns=columns,
+        use_row_groups=True,
+        gpu_memory_frac=gpu_memory_frac,
+        names=allcols_csv,
+    )
+
+    columns_ctx = {}
+    columns_ctx["all"] = {}
+    columns_ctx["all"]["base"] = columns
+
+    # Substring
+    # Replacement
+    op_seriesOp = ops.SeriesOps(
+        names="name-string",
+        names_new=None,
+        op_names=["str", "slice"],
```

Comment:
I think the syntax here is a little cumbersome for calling this operator -

How about we pass a callable instead? This will mean that we don't need to specify the op_names and other parameters:
```
ops.SeriesOp(lambda col: col.str.slice(1, 3)), ...
ops.SeriesOp(lambda col: col.str.replace("e", "XX"), ...
ops.SeriesOp(lambda col: col.astype("float"), ...
ops.SeriesOp(lambda col: col.dt.weekday(), ...
```

Also with the current syntax you can't call a function that has 'replace' as a parameter name (since the replace value will get passed to the SeriesOp rather than the kwargs)